### PR TITLE
Add methods to decode a singe char and word

### PR DESCRIPTION
**Pull request by myelin0**
**What does this PR do?**

- Add a method to decode a single character and a sentence

**Description of Task to be completed**

All tasks related to deployment are completed

### CODE REVIEW

File: decode.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                     '...-' => 'V', '.--' => 'W', '-..-' => 'X', '-.--' => 'Y', '--..' => 'Z', '.----' => '1',
+                     '..---' => '2', '...--' => '3', '....-' => '4', '.....' => '5', '-....' => '6', '--...' => '7',
+                     '---..' => '8', '----.' => '9', '-----' => '0' }
+
+# method to decode a char
+def decode_char(string)
+  @decoder_letters.fetch(string)
+end
+
+puts decode_char('.-')
```
Comment (FrederickMih) [inline decode.rb]: - [x] Based on the requirements of this project, this method should return **"A"** with quotes and not **A** without quotes as in your case.

File: decode.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  output = ''
+  string
+    .split
+    .each do |i|
+    output += decode_char(i)
+  end
+  output
+end
+
+puts decode_word('-- -.--')
```
Comment (FrederickMih) [inline decode.rb]: - [x] Hi @myelin0, if you check once more the project requirements, this method returns **"MY"** and NOT **MY**. Kindly make sure that the word is surrounded with quotes.

File: decode.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+def decode_message(_str)
+  output = []
+  string
+    .split('   ')
+    .each do |i|
+    output.push(decode_word(i))
+  end
+  output.join(' ').gsub(/\s+/, ' ')
+end
+puts decode_message('.-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...')
```
Comment (FrederickMih) [inline decode.rb]: - [x] @myelin0, I think there's a problem with this method. It throws an error. Kindly make sure that the method is working as expected.
- **Screenshot:**
Screenshot from 2022-07-20 18-18-17

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @myelin0 @bizip , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




# To Highlight

- PR has a title and summary :heavy_check_mark: 
- Linters are OK :heavy_check_mark: 
- GitHub workflow has been followed :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decode.rb (hunk 3 above)
Comment (myelin0) [reply decode.rb]: Thank you @FrederickMih for your review! I fixed the issue.

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @myelin0 and @bizip     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**